central/authprovider/registry/singleton.go

```diff
@@ -29,5 +30,9 @@ func initialize() {
 // Singleton returns the auth providers registry.
 func Singleton() authproviders.Registry {
 	once.Do(initialize)
+	// Set the auth provider registry function in the group datastore to break circular dependency
+	groupDataStore.SetAuthProviderRegistryFunc(func() authproviders.Registry {
+		return registry
+	})
```

Comment:
Could this be moved to the initialize function after the registry initialization ?